tests/production_pause_vote_timeout.py

```diff
@@ -0,0 +1,200 @@
+
+    Print("Set finalizer policy and start transition to Savanna")
+    # Specifically, need to configure finalizer name for finalizercNode as defproducerc
+    transId = cluster.setFinalizers(nodes=[node0, node1, finalizercNode], finalizerNames=["defproducera", "defproducerb", "defproducerc"])
+    assert transId is not None, "setfinalizers failed"
+    assert cluster.biosNode.waitForTransFinalization(transId), f"setfinalizers transaction {transId} was not rolled into a LIB block"
+    assert cluster.biosNode.waitForLibToAdvance(), "LIB did not advance after setFinalizers"
+
+    # biosNode no longer needed
+    cluster.biosNode.kill(signal.SIGTERM)
+    cluster.waitOnClusterSync(blockAdvancing=5)
+
+    Print("Wait for LIB on all producing nodes to advance")
+    assert node0.waitForLibToAdvance(), "node0 did not advance LIB"
+    assert node1.waitForLibToAdvance(), "node1 did not advance LIB"
+    assert producercNode.waitForLibToAdvance(), "producercNode did not advance LIB"
+
+    ####################### test 1 ######################
+
+    Print("Shutdown finalizercNode")
+    finalizercNode.kill(signal.SIGTERM)
+    assert not finalizercNode.verifyAlive(), "finalizercNode did not shutdown"
+
+    # wait some time for producercNode paused
+    paused = False
+    for i in range(0, 15):
+        time.sleep(1)
+        # Do not use waitForHeadToAdvance() to check for pausing, as producercNode
+        # still receive blocks from node0 and node1 and can make head advance
+        paused = producercNode.processUrllibRequest("producer", "paused", returnType=ReturnType.raw)
+        if paused == b'true':
+            Print(f'paused after {i} seconds after finalizercNode was shutdown')
+            break;
+    # Verify producercNode paused
+    assert paused, "producercNode still producing after finalizercNode was shutdown"
+    # Verify node0 and node1 still producing but LIB should not advance
+    assert node0.processUrllibRequest("producer", "paused", returnType=ReturnType.raw) == b'false', "node0 paused after finalizercNode was shutdown"
+    assert node1.processUrllibRequest("producer", "paused", returnType=ReturnType.raw) == b'false', "node1 paused after finalizercNode was shutdown"
+    if node0.waitForLibToAdvance(timeout=5): # LIB can advance for a few blocks first
+        assert not node0.waitForLibToAdvance(timeout=5), "LIB should not advance on node0 after finalizercNode was shutdown"
+    if node1.waitForLibToAdvance(timeout=5):
+        assert not node1.waitForLibToAdvance(timeout=5), "LIB should not advance on node1 after finalizercNode was shutdown"
+
+    Print("Restart finalizercNode")
+    finalizercNode.relaunch()
+
+    Print("Verify LIB advances after restart of finalizercNode")
+    assert node0.waitForLibToAdvance(), "node0 did not advance LIB"
+    assert node1.waitForLibToAdvance(), "node1 did not advance LIB"
+    assert producercNode.waitForLibToAdvance(), "producercNode did not advance LIB"
+
```

Comment:
You still need to check that `producercNode` automatically resumed block production.